tests/postgres_tests/test_trigram.py

```diff
@@ -149,6 +152,21 @@ def test_trigram_strict_word_distance(self):
             ],
         )
 
+    def test_trigram_concat_precedence(self):
+        """
+        Trigram functions should be wrapped in parentheses to be safe
+        as if one used CONCAT(...)
+        """
+        search_term = "Mat"
+        self.assertSequenceEqual(
+            self.Model.objects.annotate(
+                concat_result=Concat(F("field"), Value("tew")),
```

Comment:
```suggestion
                concat_result=Concat(
                    F("field"), Value("tew"), output_field=self.model._meta.get_field("field"),
```

Reply:
Does this add a `(...)::text` to the mix? it was missing that output_field that the bug happened but if the result of having output_field implies the existence of that cast, then the bug does not happen

Reply:
How about setting the `output_field` on the `Value`?